modules/exploits/linux/http/vmware_vrni_rce_cve_2023_20887.rb

```diff
@@ -0,0 +1,134 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'VMWare Aria Operations for Networks (vRealize Network Insight) pre-authenticated RCE',
+        'Description' => %q{
+          VMWare Aria Operations for Networks (vRealize Network Insight) is vulnerable to command injection when accepting user input through the Apache Thrift RPC interface. This vulnerability allows a remote unauthenticated attacker to execute arbitrary commands on the underlying operating system as the root user. The RPC interface is protected by a reverse proxy which can be bypassed. VMware has evaluated the severity of this issue to be in the Critical severity range with a maximum CVSSv3 base score of 9.8.
+          a malicious actor can get remote code execution in the context of 'root' on the appliance.
+          VMWare 6.x version are vulnerable.
+
+          This module exploits the vulnerability to upload and execute payloads gaining root privileges.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Sina Kheirkhah', # Metasploit Module (@SinSinology) of Summoning Team (@SummoningTeam) on twitter
+        ],
+        'References' => [
+            ['CVE', 'CVE-2023-20887'],
+            ['URL', 'https://www.vmware.com/security/advisories/VMSA-2023-0012.html'],
+            ['URL', 'https://summoning.team/blog/vmware-vrealize-network-insight-rce-cve-2023-20887/'],
+        ],
+        'DisclosureDate' => '2023-06-07',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
```

Comment:
```suggestion
        'Arch' => [ARCH_CMD, ARCH_X64],
```
Should `ARCH_x86` be removed? I couldn't find out whether or not 32-bit architecture is supported by vRealize Network Insights after a quick google. I noticed your Linux Dropper target doesn't specify `ARCH_X86`